.github/workflows/build.yml

```diff
@@ -1,301 +1,293 @@
-          -   name: Compare benchmark result and make comment
-              uses: benchmark-action/github-action-benchmark@v1
+                  flags: UnitTests
+                  token: ${{ secrets.CODECOV_TOKEN }}
+              if: ${{ matrix.node == 18 }}
+
+    integration:
+        name: Integration # (ganache with HTTP)
+        needs: build
+        runs-on: ubuntu-latest
+        env:
+            INFURA_GOERLI_HTTP: ${{ secrets.INFURA_GOERLI_HTTP }}
+            INFURA_GOERLI_WS: ${{ secrets.INFURA_GOERLI_WS }}
+        strategy:
+            fail-fast: false
+            matrix:
+                node: [18]
+                backend: ['ganache']
+                mode: ['http']
+        steps:
+            - uses: actions/setup-node@v4
               with:
-                  # What benchmark tool the output.txt came from
-                  tool: 'benchmarkjs'
-                  # Where the output from the benchmark tool is stored
-                  output-file-path: benchmark-data.txt
-                  # Where the previous data file is stored
-                  external-data-json-path: web3-benchmark-main.json
-                  # Workflow will fail when an alert happens
-                  fail-on-alert: false
-                  # GitHub API token to make a commit comment
-                  github-token: ${{ secrets.GITHUB_TOKEN }}
-                  # Enable alert commit comment
-                  comment-always: true
-                  save-data-file: false
-              # copy comment from commit to Pull Request
-          -   run: node scripts/copyCommitCommentToPrComment.js ${{ secrets.GITHUB_TOKEN }} ${{github.event.pull_request.head.sha}} ${{github.event.number}}
-          -   name: Compare benchmark result and fail if threshold is reached
-              uses: benchmark-action/github-action-benchmark@v1
+                  node-version: ${{ matrix.node }}
+            - uses: actions/cache/restore@v3
               with:
-                  # What benchmark tool the output.txt came from
-                  tool: 'benchmarkjs'
-                  # Where the output from the benchmark tool is stored
-                  output-file-path: benchmark-data.txt
-                  # Where the previous data file is stored
-                  external-data-json-path: web3-benchmark-main.json
-                  # Workflow will fail when an alert happens
-                  fail-on-alert: true
-                  # Enable alert commit comment
-                  alert-threshold: '100%'
-                  comment-always: false
-          -   name: Save main branch benchmark data
-              uses: actions/cache/save@v3
-              if: github.event_name == 'push' && github.ref == 'refs/heads/4.x'
+                path: ./
+                key: web3-${{ matrix.node }}-${{github.event.pull_request.base.sha}}
+            - run: yarn test:e2e:${{ matrix.backend }}:${{ matrix.mode }}
+              shell: bash
+
+    e2e-ipc:
```

Comment:
its for all transports 'ipc', 'ws', 'http' , so i'll suggest to rename job instead of e2e-ipc